array.c

```diff
@@ -935,21 +935,26 @@ rb_check_to_array(VALUE ary)
 
 /*
  *  call-seq:
- *     Array.try_convert(obj) -> array or nil
+ *    Array.try_convert(obj) -> new_array or nil
  *
- *  Tries to convert +obj+ into an array, using the +to_ary+ method.  Returns
- *  the converted array or +nil+ if +obj+ cannot be converted.
- *  This method can be used to check if an argument is an array.
+ *  Tries to convert <tt>obj</tt> to an \Array.
  *
- *     Array.try_convert([1])   #=> [1]
- *     Array.try_convert("1")   #=> nil
+ *  When <tt>obj</tt> is an
+ *  {Array-convertible object}[doc/implicit_conversion_rdoc.html#label-Array-Convertible+Objects],
+ *  returns the \Array object created by converting it:
  *
- *     if tmp = Array.try_convert(arg)
- *       # the argument is an array
- *     elsif tmp = String.try_convert(arg)
- *       # the argument is a string
- *     end
+ *    class ToAryReturnsArray < Set
+ *      def to_ary
+ *        self.to_a
+ *      end
+ *    end
+ *    as = ToAryReturnsArray.new([:foo, :bar, :baz])
+ *    Array.try_convert(as) # => [:foo, :bar, :baz]
  *
+ *  Returns <tt>nil</tt> unless <tt>obj</tt> is an
+ *  {Array-convertible object}[doc/implicit_conversion_rdoc.html#label-Array-Convertible+Objects]:
```

Comment:
Do we need the link twice?

How about "Returns +nil+ if +obj+ is not Array-convertible." which also makes the negative easier to follow for non-native English readers.

Reply:
Done, throughout.